packages/authorization-process/src/model/domain/apiConverter.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Client,
+  ClientKind,
+  Key,
+  KeyUse,
+  clientKind,
+  keyUse,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import {
+  ApiClient,
+  ApiClientWithKeys,
+  ApiClientKind,
+  ApiKey,
+  ApiKeyUse,
+} from "./models.js";
+
+export const ClientKindToApiClientKind = (kind: ClientKind): ApiClientKind =>
+  match<ClientKind, ApiClientKind>(kind)
+    .with(clientKind.consumer, () => "CONSUMER")
+    .with(clientKind.api, () => "API")
+    .exhaustive();
+
+export const KeyUseToApiKeyUse = (kid: KeyUse): ApiKeyUse =>
+  match<KeyUse, ApiKeyUse>(kid)
+    .with(keyUse.enc, () => "ENC")
+    .with(keyUse.sig, () => "SIG")
+    .exhaustive();
+
+export const clientToApiClient = (
+  client: Client,
+  withKeys: boolean
+): ApiClient | ApiClientWithKeys => ({
+  id: client.id,
+  name: client.name,
+  consumerId: client.consumerId,
+  users: client.users,
+  createdAt: client.createdAt.toJSON(),
+  purposes: client.purposes,
+  kind: ClientKindToApiClientKind(client.kind),
+  description: client.description ? client.description : undefined,
+  ...(withKeys ? { keys: [client.keys] } : {}),
+});
+
+export const keyToApiKey = (key: Key): ApiKey => ({
+  name: key.name,
+  createdAt: key.createdAt.toJSON(),
+  kid: key.kid,
+  encodedPem: key.encodedPem,
+  algorithm: key.algorithm,
+  use: KeyUseToApiKeyUse(key.use),
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  userId: key.userId!, // TODO Double check
```

Comment:
In the `keys.ts `model there was not the `userId` but the `relatshionshipId`, are they the same thing?
But above all, since both are optional, what do we put in cases where they are not present?
I expect that at least one of the two will always be there, but if not, I saw from a scale route that the `Kid` wears it, is it possible?

Reply:
They are not the same, if `userId` is missing it should throw `MissingUserId` otherwise you just assign it to `userId`